pkg/virt-controller/watch/migration.go

```diff
@@ -472,33 +465,20 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 	return nil
 }
 
-func (c *MigrationController) createTargetPod(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance) error {
+func (c *MigrationController) createTargetPod(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance, attachmentPod *k8sv1.Pod) error {
 
-	templatePod, err := c.templateService.RenderLaunchManifest(vmi)
+	templatePod, err := c.templateService.RenderLaunchManifest(vmi, attachmentPod)
 	if err != nil {
 		return fmt.Errorf("failed to render launch manifest: %v", err)
 	}
 
-	antiAffinityTerm := k8sv1.PodAffinityTerm{
-		LabelSelector: &v1.LabelSelector{
-			MatchLabels: map[string]string{
-				virtv1.CreatedByLabel: string(vmi.UID),
-			},
-		},
-		TopologyKey: "kubernetes.io/hostname",
-	}
-	antiAffinityRule := &k8sv1.PodAntiAffinity{
-		RequiredDuringSchedulingIgnoredDuringExecution: []k8sv1.PodAffinityTerm{antiAffinityTerm},
-	}
-
-	if templatePod.Spec.Affinity == nil {
-		templatePod.Spec.Affinity = &k8sv1.Affinity{
-			PodAntiAffinity: antiAffinityRule,
+	// No attachment pod means we the attachment pod is the first pod to
+	// create in the migration process, so we need to set its affinities.
```

Comment:
vmi pod must be the first pod to be created. The VMI pod has all the device/cpu/mem requirements necessary to run the VMI in the correct location.